java-util/src/main/java/io/druid/java/util/common/granularity/AllGranularity.java

```diff
@@ -38,14 +38,7 @@
   private final DateTime maxDateTime = new DateTime(MAX_INSTANT);
   private final DateTime minDateTime = new DateTime(MIN_INSTANT);
 
-  private static final AllGranularity INSTANCE = new AllGranularity();
-
-  private AllGranularity() {}
-
-  public static AllGranularity getInstance()
-  {
-    return INSTANCE;
-  }
+  public AllGranularity() {}
```

Comment:
I don't think there's a point in even making those package private.  There is no such thing as a "semantically it is a singleton".  Singleton-ness is a concept that exists when a user decides to only have a single instance that is reused everywhere in their code.  It is not something that the person who is writing the class can or should make a decision about.  The constructor here being public is not going to cause anyone to do anything incorrectly or cause incorrect functionality if it is used.